lib/auth/Vault.ts

```diff
@@ -24,6 +24,9 @@ function vaultSignatureCb(
     if (err) {
         log.debug('received error message from auth provider',
             { errorMessage: err });
+        if (err.code && err.code === 'InvalidAccessKeyId') {
+            return callback(errors.InvalidAccessKeyId);
+        }
```

Comment:
Can we also try this change in cloudserver (I guess cloudserver CI should be enough, but because we don't really use vault there, maybe Integration/Zenko are needed) - just to be on the safe side, so we do not have any issue when the function returns an arsenal error

Reply:
That's a good point. Not sure if we test this code path, I will check.

Reply:
I feel like the issue we are facing is that we are sometimes managing Arsenal errors and other times 'request' errors, depending on the type of error and the backend being used. Should we consider translating 'request' errors into Arsenal errors so that we only need to manage one type of error?

Reply:
Yes. Adding another note: we recently moved out of the arsenal errors in vaultclient, because we were importing the repo only to use these Arsenal errors (based on some metaprogramming logic, with perf impact). Maybe this is a more global thing to discuss here...